packages/nostr/src/client/index.ts

```diff
@@ -216,19 +202,39 @@ export class Nostr {
   /**
    * Publish an event.
    */
-  async publish(event: Event): Promise<void> {
+  async publish(event: Event, key: PrivateKey): Promise<void> {
```

Comment:
We wont always have the private key so we need a way to publish events that are already signed